packages/components/src/truncate/types.ts

```diff
@@ -1,14 +1,26 @@
-export type TruncateEllisizeMode = 'auto' | 'head' | 'tail' | 'middle' | 'none';
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
 
-export interface Props {
+export type TruncateEllipsizeMode =
+	| 'auto'
+	| 'head'
+	| 'tail'
+	| 'middle'
+	| 'none';
+
+export type TruncateProps = {
 	/**
 	 * The ellipsis string when `truncate` is set.
```

Comment:
As with the `limit` prop, the `truncate` this description is referencing is the `truncate` prop on components that are _based on_ `<Truncate>` (e.g. `<Text>`, `<Heading>`, maybe more). So it's nonsensical as a description in `<Truncate>` itself.

In light of the behavior quirk that Marco [found](https://github.com/WordPress/gutenberg/pull/41697/files/6b6dc296dacf38cc8d21ed7965c689e2a49b9c6b#r898029123), and to avoid having to override these descriptions for the Truncate-based components, maybe we can say "The ellipsis string when truncating the text by a `limit` value"?

@ciampo Aside: Maybe this is another instance where grouping the pass-through props together is a good idea. Like for the `Heading` props doc, we could've just had one prop `truncateProps` with the description "Settings for truncating the text. See the `Truncate` documentation for details.".

Reply:
> maybe we can say "The ellipsis string when truncating the text by a limit value"?

Sounds good to me, maybe to be clear we can say "the `limit` prop's value" ?

> Maybe this is another instance where grouping the pass-through props together is a good idea. Like for the Heading props doc, we could've just had one prop truncateProps with the description "Settings for truncating the text. See the Truncate documentation for details.".

This is definitely worth considering, yes! I'd like to take a wider look at the package, though, and understand if this is a good pattern that we can adopt throughout the codebase (or if, for example, there are better alternatives). Ultimately, I'd love it if the components had consistent APIs for composition